public/js/modules/signupFunctions.js

```diff
@@ -1,22 +1,19 @@
-const validate = (field, regex, cb => {
+const validate = (field, regex, cb) => {
     let notAccepteable = [];
     const keys = Object.keys(field)
     for (let i = 0; i < keys.length; i++) {
-        if (!field[keys[i]].match(regex)) {
-            unaccepteable.push(field[key]);
+        if (!regex[i].test(field[keys[i]])) {
+            notAccepteable.push(keys[i]);
         }
     }
     if (notAccepteable.length > 0) {
-         swal(text: `make sure you fill the following fields ${unaccepteable.join(',')}`, type: "warning");
-         return cb(false);
+        swal(text: `make sure you fill the following fields ${unaccepteable.join(',')}`, type: "warning");
+        return cb(false);
     }
     return cb(true);
-});
-
+};
```

Comment:
@Anies12 This isn't a super urgent issue, but since this validate function isn't doing anything asynchronous (it's not doing a fetch or waiting on anything like that), it could just return it's result directly instead of requiring a `cb` function to be passed in.  That might make it a little easier to use (cb functions sometimes need to have bind() used, if they're inside a class instance, for example, so you would avoid that headache).
Then you'd just use it like this: 
```
if (validate(secondStepfields, secondStepValidationRegex) {
  ToggleDisplay
  ...
```